ci/travis/check_spelling.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -e
+
+echo "Spell check"
+
+if [[ !  -z  $TRAVIS_PULL_REQUEST_BRANCH  ]]; then
+  # if on a PR, just analyse the changed files
+  echo "TRAVIS PR BRANCH: $TRAVIS_PULL_REQUEST_BRANCH"
+  FILES=$(git diff --name-only HEAD $(git merge-base HEAD master) | tr '\n' ' ')
+  echo "Running on edited files: $FILES"
+else
+  FILES="."
+fi
+
+           
+RE=$(cut -d: -f1 scripts/spelling.dat |  tr '\n' '\|' | sed -e 's/|$//')
+unbuffer ag --smart-case --all-text --nopager --numbers --word-regexp --path-to-ignore ci/travis/.agignore "$RE" $FILES | tee spellcheck.out
+
+if [ ! -f spellcheck.out ]; then
+  exit 0
+else
+  echo "Spelling errors have been found"
+  exit 1
+fi
+
```

Comment:
Even then it would be better for me if I get the test results when I do `git commit` and not having to wait for a roundtrip me -> github -> travis -> me.